### DISCUSSION TITLE
Does quarkus native image support dynatrace introspection agent?

### DISCUSSION THREAD

**Discussion by sivakrishnak1** [quarkusio/quarkus#39655] [Q&A]
We are observing an issue with dynatrace introspection packages (com.dyntrace.*) with building quarkus native image for application.
It reports:

Detected an instance of Random/SplittableRandom class in the image heap. Instances created during image generation have cached seed values and don't behave as expected.

when I tried to exclude this with initialization at runtime property, build was succesful but after deploying to kubernetes cluster, observing pod restart issues and eventually deployent is getting stuck.

Looking for inputs on the same.

[a comment by an automated account is not shown]

**Reply by maxandersen** [reply]
I wasn't aware dynstrace had a native image compatible agent. Do you have some info on what/how you are setting it up?

**Reply by sivakrishnak1** [reply]
@maxandersen even i dont have enough info on dynatrace compatible agent, its just have come across this issue and was looking for help. This package is causing native build image failure and somehow being reference in the final native image that is getting created.

**Reply by maxandersen** [reply]
Ok so afaik dynatrace haven't made a version available that works with native image. Best to raise it to dynatrace as their code is closed so we can't really help much without them being involved.

**Comment by brunobat** [comment]
@sivakrishnak1 do you have more details on where to find this `Random/SplittableRandom`?

**Reply by geoand** [reply]
Right, I think GraalVM provides some information.

**Reply by sivakrishnak1** [reply]
com.dynatrace.agent.introspection.uem.UemsessionGenerator is the rootcause of this issue.

**Reply by zakkak** [reply]
Hi @sivakrishnak1 , you might find this https://foivos.zakkak.net/tutorials/working-with-randoms-native-images/ useful (see the last section).
